### PR TITLE
[BE] Get rid of `six` in caffe2 code

### PR DESCRIPTION
**Pull request by malfet**
Mostly `s/string_types/str/` `s/binary_types/bytes/` and `s/text_types/str/`
Also `y.extend([str(x) for x in foo])`->`y.extend(map(str, foo))`
As Python-2 is long dead

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks!

**Review by Skylion007 — APPROVED** [review]

**Comment by Skylion007** [comment]
I can't wait to finally remove six from the pytorch requirements.

File: .circleci/docker/common/install_conda.sh [hunk 1]
```diff
@@ -75,7 +75,7 @@ if [ -n "$ANACONDA_PYTHON_VERSION" ]; then
   }
 
   # Install PyTorch conda deps, as per https://github.com/pytorch/pytorch README
-  CONDA_COMMON_DEPS="astunparse pyyaml mkl=2021.4.0 mkl-include=2021.4.0 setuptools six"
+  CONDA_COMMON_DEPS="astunparse pyyaml mkl=2021.4.0 mkl-include=2021.4.0 setuptools"
```
Comment (Skylion007) [inline .circleci/docker/common/install_conda.sh:78]: Still one left in functorch/examples/maml_omniglot/support/omniglot_loaders.py and what about torch._six?

File: caffe2/python/core.py [hunk 2]
```diff
@@ -385,11 +384,11 @@ def CreateOperator(
     # Add rectified inputs and outputs
     inputs = _RectifyInputOutput(inputs)
     outputs = _RectifyInputOutput(outputs)
-    operator.input.extend([text_type(i) for i in inputs])
-    operator.output.extend([text_type(o) for o in outputs])
+    operator.input.extend([str(i) for i in inputs])
+    operator.output.extend([str(o) for o in outputs])
```
Comment (Skylion007) [inline caffe2/python/core.py:388]:
```suggestion
    operator.input.extend(map(str, inputs))
    operator.output.extend(map(str,outputs))
```
while we are here, might as well make it more efficient.

File: caffe2/python/core.py [hunk 3]
```diff
@@ -385,11 +384,11 @@ def CreateOperator(
...
     if control_input:
         control_input = _RectifyInputOutput(control_input)
-        operator.control_input.extend([text_type(i) for i in control_input])
+        operator.control_input.extend([str(i) for i in control_input])
```
Comment (Skylion007) [inline caffe2/python/core.py:391]: same here

File: .circleci/docker/common/install_conda.sh (hunk 1 above)
Comment (malfet) [reply .circleci/docker/common/install_conda.sh:78]: `torch._six` can stay, though we probably can eradicate it at some point

File: caffe2/python/core.py (hunk 2 above)
Comment (malfet) [reply caffe2/python/core.py:388]: Are you aware of the perf implications? I remember `map` used to be slower than list comprehensions in 2.7 days.

File: .circleci/docker/common/install_conda.sh (hunk 1 above)
Comment (malfet) [reply .circleci/docker/common/install_conda.sh:78]: Also,`examples` might have dependencies that are not covered by CI/CD nor by package dependencies

File: caffe2/python/core.py (hunk 2 above)
Comment (malfet) [reply caffe2/python/core.py:388]:
```
>>> timeit.timeit('[str(x) for x in range(100)]')
9.269361833000005
>>> timeit.timeit('list(map(str, range(100)))')
8.176639707999982
```
Ok, map is faster

**Comment by Skylion007** [comment]
BTW @malfet when can we get rid of the python-six submodule in thirdparty?

**Comment by malfet** [comment]
> BTW @malfet when can we get rid of the python-six submodule in third_party?

@Skylion007 propose a PR and I'll stamp it 👍

**Comment by Skylion007** [comment]
oof, we missed a few places. Here and in pyproject.toml

**Pull request closed without merging**